### PR TITLE
Move Core Font Installer Notice to selected pages.

### LINKED ISSUE #1439: Change where Core Font Installer Notice is shown
**Description**
Currently, the Core Font Installer notice is shown on all WordPress admin pages outside of Gravity Forms, and any Gravity PDF pages. To reduce notice fatigue for users, only display the notice on the WordPress Plugin screen, all Gravity PDF pages, and the following Gravity Forms screens:

1. Form List
2. Form Settings
3. Form Entry List
4. Form Entry Details

**Screenshots**
Screenshot 2022-12-19 at 11 59 23 am

### PR DESCRIPTION
**Pull request by jestonihpi**
## Description
WIP. I have only set the  Core Font Installer Notice exclusively to Gravity PDF pages. I need to add this to these pages as well:

1. Form List
2. Form Settings
3. Form Entry List
4. Form Entry Details

#1439 
## Checklist:
- [ ] I've tested the code.
- [ ] My code is easy to read, follow, and understand
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation / docblocks.

### CODE REVIEW

**Comment by jakejackson1** [comment]
@jestonihpi I think you've misunderstood. We aren't removing the notice from any existing pages. We are only enabling it on these specific GF pages:


    Form List
    Form Settings
    Form Entry List
    Form Entry Details

**Comment by jestonihpi** [comment]
Hi @jakejackson1, can you review the recent changes on this PR please?

This would look like on Form list:
image
The font will be reset to adapt to the current error messages.

But when its on Gravity PDF pages, it will look like this:
image

Not sure if this is a good compromise, please advise.

Thanks!

File: src/Controller/Controller_Actions.php [hunk 1]
```diff
@@ -173,8 +179,20 @@ public function route_notices() {
...
+				$message = call_user_func( $route['view'], $route['action'], $route['action_text'] );
+				/* Lets load the Install Core font message to Helper_Notice if the current page belongs to Gravity PDF or the plugin directory. */
+				if ( $is_gfpdf_page || $pagenow === 'plugins.php' ) {
+					$class = ( isset( $route['view_class'] ) ) ? $route['view_class'] : '';
+					$this->notices->add_notice( $message, $class );
+
+				} else {
+					/* Add  Install Core font message GFCommon if it doesn't. Remove the font styling to match with the generic add-on error message. */
+					if ( ! empty( \GFCommon::$errors ) ) {
+						$message = '<br>' . preg_replace( '/(<[^>]+) style=".*?"/i', '$1', $message );
```
Comment (jakejackson1) [inline src/Controller/Controller_Actions.php]: Remove this preg_replace. We can probably always wrap message in a `<p>` or `<div>` tag if the message displays inline with an existing message.

File: src/Controller/Controller_Actions.php [hunk 2]
```diff
@@ -151,10 +151,16 @@ public function get_routes() {
...
 
+		global $pagenow;
 		/* Prevent actions being displayed on our welcome pages */
-		if ( ! is_admin() ||
-			 ( rgget( 'page' ) === 'gfpdf-getting-started' ) || ( rgget( 'page' ) === 'gfpdf-update' ) || ( defined( 'DOING_AJAX' ) && DOING_AJAX )
-		) {
+		$register_routes = false;
+		$is_gfpdf_page   = \GPDFAPI::get_misc_class()->is_gfpdf_page();
+
+		if ( ! wp_doing_ajax() && is_admin() && ( $pagenow === 'plugins.php' || $is_gfpdf_page || \GFForms::is_gravity_page() ) ) {
```
Comment (jakejackson1) [inline src/Controller/Controller_Actions.php]: As I mentioned, assign all conditions to individual variables so this IF statement is easy to understad.

**Review by jakejackson1 — CHANGES_REQUESTED** [review]
@jestonihpi just a few minor corrections needed.

**Comment by jakejackson1** [comment]
Resolved in https://github.com/GravityPDF/gravity-pdf/pull/1549

**Pull request closed without merging**